paddle/phi/core/ddim.cc

```diff
@@ -172,10 +172,13 @@ DDim stride_numel(const DDim& ddim) {
 }
 
 DDim DDim::reshape(std::vector<int>& shape) const {
```

Comment:
I am not sure about these changes to this file. I see what you did here, but after your changes errors associated with incorrect dimensions will result in something like segfault while earlier we were getting framework-handled errors with nice description, what do you think about that?

Reply:
Both `reshape()` and `transpose()` in ddim.cc were added for purpose of matmul fuse passes. As we check dimensions with PADDLE_ENFORCES during fusion, there is no possibility that values here could be incompatible.

To avoid possible misunderstandings, maybe we should move these functions outside phi/core to some mkldnn_helper function?

Reply:
Moving it outside phi/core is a great idea! Except from that little issue, LGTM! Really good work(as always!)